### ISSUE TITLE
install/run instructions?

### ISSUE THREAD

**Issue by kurokirasama** [mathworks/MATLAB-language-server#27]
Hey, sorry my ignorance, but how to run or install this? There is anything in the readme file

**Comment by dklilley** [comment]
Hi @kurokirasama , thank you for raising this question! There is a known lack of documentation around the development workflows. This is a space we want to build up in the future.

The primary commands are the following:
* `npm install` - installs all needed NPM packages
* `npm run compile` - build the TypeScript code into executable JavaScript code for development
* `npm run package` - similar to `npm run compile`, but performs a production build

When it comes to running the language server, I believe this can be done with `node out/index.js`. The possible startup flags are enumerated in [ConfigurationManager.ts](https://github.com/mathworks/MATLAB-language-server/blob/main/src/lifecycle/ConfigurationManager.ts):
https://github.com/mathworks/MATLAB-language-server/blob/319868d5701d01437aeb1c8d82f2803a48daf995/src/lifecycle/ConfigurationManager.ts#L8-L14

The descriptions for these can be found within [CliUtils.ts](https://github.com/mathworks/MATLAB-language-server/blob/main/src/utils/CliUtils.ts).


Please feel free to reach out if you have any other questions about this.

**Comment by kurokirasama** [comment]
Thanks a lot! As I understand, this need matlab 2021+, right?

**Comment by dklilley** [comment]
Yes, that is correct - the language server currently support MATLAB R2021a and later.

**Comment by kurokirasama** [comment]
does it needs polyspace?

**Comment by dklilley** [comment]
No, it does not require Polyspace. All features provided by this language server should only require a base MATLAB installation.

**Comment by kurokirasama** [comment]
super! Thanks!

**Comment by kurokirasama** [comment]
can you provide some guidance in how to set up this in sublime text?

**Comment by dklilley** [comment]
The `matlabInstallPath` flag should be set to something like `/usr/local/MATLAB/R2023a` (or the value returned by `matlabroot` when executed within MATLAB.

The `matlabLaunchCommandArgs` flag is for providing additional startup options to MATLAB, but this is not likely to be necessary.

Regarding Sublime Text, I will respond on your other issue (#29)

**Comment by gregdvd** [comment]
Requirements would also be nice - I fought for a couple of days with the "Client X quit with exit code 1 and signal 0." error message before discovering that it was caused by my "node" and "npm" being too old. I was using the version provided by my Ubuntu through apt (8.5.1) - now, after installing "npm 10.5.0" directly from nodejs.org, everything works fine.

BTW, thanks a lot for this project! It makes editing matlab files much easier!
